### DISCUSSION TITLE
Plans to support Svelte 5 when it drops (likely in April)?

### DISCUSSION THREAD

**Discussion by gchartier** [xyflow/xyflow#3940] [Svelte]
Hello,

This library is just fantastic and I'm excited to build with it in the next few months!

I primarily use Svelte and the release of Svelte 5 is on the horizon. In fact, I think it's likely to be released at the Svelte Summit on April 27th. Since it's a fairly large change for Svelte, I'm wondering if you all will be prioritizing support for Svelte 5 when it is finally released / if you are already preparing for it.

I imagine it will change the DX since stores are replaced by $state and $derived runes among other API changes that would affect the svelte-flow APIs.

Thank you for your work on this library and keep it up!!

I may jump in with some open source contributions once I get in the weeds of my project. All the best 💠

**Comment by moklick** [comment]
Hey @gchartier 

we are currently busy with the next major release for React Flow. After that we might start to rewrite Svelte Flow to Svelte 5. Our plan was to wait until Svelte 5 is stable but it seems to be pretty much the case now so I don't see any reason to wait any longer. 

I will keep this issue open until we have a branch for Svelte 5 and start working on it.

**Reply by gchartier** [reply]
Awesome!

I believe this library is such an important piece of the future of an open-source Personal Knowledge Management ecosystem.

Thanks again for your work.

**Comment by Superd22** [comment]
Hey, first of all thanks for the great library !


Until there's official support for svelte 5, I've been able to use the library with a couple work-arounds.

1. packages/svelte/src/lib/components/NodeWrapper/NodeWrapper.svelte needs to be patched
This can be achieve with a tool like [patch-package](https://www.].com/package/patch-package)
```diff
diff --git a/node_modules/@xyflow/svelte/dist/lib/components/NodeWrapper/NodeWrapper.svelte b/node_modules/@xyflow/svelte/dist/lib/components/NodeWrapper/NodeWrapper.svelte
index 8169b29..d086aa3 100644
--- a/node_modules/@xyflow/svelte/dist/lib/components/NodeWrapper/NodeWrapper.svelte
+++ b/node_modules/@xyflow/svelte/dist/lib/components/NodeWrapper/NodeWrapper.svelte
@@ -98,6 +98,7 @@
     // if type, sourcePosition or targetPosition changes,
     // we need to re-calculate the handle positions
     const doUpdate =
+      (prevNodeRef !== nodeRef) ||
       (prevType && nodeType !== prevType) ||
       (prevSourcePosition && sourcePosition !== prevSourcePosition) ||
       (prevTargetPosition && targetPosition !== prevTargetPosition);
```

2. In case of SSR issues while using sveltekit (ERR_UNSUPPORTED_DIR_IMPORT), this can be worked around without disabling SSR using the technique [described here](https://vike.dev/broken-npm-package) 

```javascript
// vite.config.js

export default {
  ssr: {
    noExternal: ['@xyflow/svelte']
  }
}
```


Again just to emphasize, those are **work-arounds** until proper support exists.

**Reply by Superd22** [reply]
@moklick I'd have loved to contribute towards proper support, but first of all i don't really understand _how_ the nodeRef's been affected by svelte 5, and secondly i figured e2e testing of both svelte 4 & 5 might not be trivial

**Comment by tsdexter** [comment]
Now that `@xyflow/react@12` is out, any updates here?

**Comment by tsdexter** [comment]
@moklick any updates on svelte 5/runes support?

**Comment by tsdexter** [comment]
https://svelte.dev/blog/svelte-5-is-alive 🎉

**Reply by JustKira** [reply]
Yaa i can't wait for svelte flow support Svelte 5

**Reply by LittleSaya** [reply]
Current version of Svelte Flow looks functional under svelte 5, but it would be much nicer if custom nodes could be written in runes mode.

**Reply by laelhalawani** [reply]
I'm getting errors with CustomNodes on SveltKit with Svelte5 using the examples from the library.

**Comment by laelhalawani** [comment]
Is there any time-frame for when you expect to drop the Svelte 5 version?

**Comment by oneezy** [comment]
looks like @peterkogo has been doing some recent work in the [migrate/svelte5](https://github.com/xyflow/xyflow/tree/migrate/svelte5) branch. could you give us an update Pete or if you need help on anything? 

can view the latest commits _(11-13-2025)_ here: 
https://github.com/xyflow/xyflow/commits/migrate/svelte5

**Reply by peterkogo** [reply]
I was out of office and sick for the past 2 weeks. I hope to release the first preliminary version this year :)

**Reply by tsdexter** [reply]
Hope you’re doing well! Good to hear about the preliminary 🙏

**Reply by oneezy** [reply]
sorry to hear you been sick bro! i actually just got sick today 🤧 

anyways, let us know if you need any help!

**Comment by JustKira** [comment]
Are there any updates or an expected timeline for the release of a beta version for migrating from Svelte 4 to Svelte 5?

**Reply by moklick** [reply]
you can follow the progress here https://github.com/xyflow/xyflow/pull/4878 - there is no timeline for a stable release but we plan a first pre-release for the next week.

**Reply by JustKira** [reply]
Sounds Great can't wait to have my hands on the pre-release

**Reply by oneezy** [reply]
sweet!!

**Comment by peterkogo** [comment]
@gchartier @JustKira @oneezy @tsdexter @laelhalawani @LittleSaya  @Superd22 
Pre-release of Svelte Flow 1.0 is out.

Some preliminary notes:
- install via `npm install @xyflow/svelte@next`
- requires Svelte 5
- take a look at the [migration guide](https://next.svelteflow.dev/learn/troubleshooting/migrate-to-v1)
- early, incomplete docs available at [next.svelteflow.dev](https://next.svelteflow.dev)

Please keep in mind that this is an alpha version and we are still in the process of migrating our API references and guides. *However,* [all examples](https://next.svelteflow.dev/examples) are already up to date!

We need your feedback! Are your experiencing any issues? What do you think about the new API?
Share your thoughts in our [Discord](https://discord.gg/RVmnytFmGW)!

**Reply by JustKira** [reply]
LET'S GOOO

**Reply by oneezy** [reply]
YESSSSS
